src/languages/r.js

```diff
@@ -2,59 +2,105 @@
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\(/, end: /\)(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\{/, end: /\}(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r"(-*)\[/, end: /\](-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\(/, end: /\)(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\{/, end: /\}(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /r'(-*)\[/, end: /\](-*)'/ }),
+          {begin: '"', end: '"', relevance: 0},
+          {begin: "'", end: "'", relevance: 0}
+        ],
+        relevance: 10
+      },
+
       {
         begin: IDENT_RE,
         keywords: {
           $pattern: IDENT_RE,
           keyword:
-            'function if in break next repeat else for return switch while try tryCatch ' +
-            'stop warning require library attach detach source setMethod setGeneric ' +
-            'setGroupGeneric setClass ...',
+            'function if in break next repeat else for while',
           literal:
-            'NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 ' +
-            'NA_complex_|10'
+            'NULL NA TRUE FALSE Inf NaN NA_integer_|10 NA_real_|10 ' +
+            'NA_character_|10 NA_complex_|10',
+          built_in:
+            // Builtin constants
+            'LETTERS letters month.abb month.name pi T F ... ..1 ..2 ' +
+            // Primitive functions
+            // These are all the functions in `base` that are implemented as a
+            // `.Primitive`, minus those functions that are also keywords.
+            'abs acos acosh all any anyNA Arg as.call as.character' +
+            'as.complex as.double as.environment as.integer as.logical' +
+            'as.null.default as.numeric as.raw asin asinh atan atanh attr' +
+            'attributes baseenv browser c call ceiling class Conj cos cosh' +
+            'cospi cummax cummin cumprod cumsum digamma dim dimnames' +
+            'emptyenv exp expression floor forceAndCall gamma gc.time' +
+            'globalenv Im interactive invisible is.array is.atomic is.call' +
+            'is.character is.complex is.double is.environment is.expression' +
+            'is.finite is.function is.infinite is.integer is.language' +
+            'is.list is.logical is.matrix is.na is.name is.nan is.null' +
+            'is.numeric is.object is.pairlist is.raw is.recursive is.single' +
+            'is.symbol lazyLoadDBfetch length lgamma list log max min' +
+            'missing Mod names nargs nzchar oldClass on.exit pos.to.env' +
+            'proc.time prod quote range Re rep retracemem return round' +
+            'seq_along seq_len seq.int sign signif sin sinh sinpi sqrt' +
+            'standardGeneric substitute sum switch tan tanh tanpi tracemem' +
+            'trigamma trunc unclass untracemem UseMethod xtfrm',
         },
         relevance: 0
       },
+
+      hljs.COMMENT(
+        '#',
+        '$',
+        {
+          returnBegin: true,
```

Comment:
If you want to match ROXYGEN you should just match it, not this match comment then return to begin.

Reply:
Ah, actually it seems I misunderstand the usage of that tag based on my reading of the C# lexer. The Java lexer interprets it completely differently from the former: C# treats the comment start (`///`) as a doctag. Java doesn’t, instead it treats actual doxygen tags (`@param` …) as doctags.

I’ll emulate Java, not C#.

Reply:
Oh, yes if you wanted the HTML output to show the nesting you'd need that, but that doesn't seem correct either. :)  Typically (in my understanding) "doctags" are something that appear "inside" of comments.

```
/** this is a comment
@type {string}  This is a doctag, etc.
*/ ditto
```